src/staticanalysis/bot/static_analysis_bot/clang/format.py

```diff
@@ -45,53 +29,25 @@ class ClangFormat(object):
     from a patch
     '''
     def __init__(self):
-        self.binary = os.path.join(
-            os.environ['MOZBUILD_STATE_PATH'],
-            'clang-tools', 'clang-tidy', 'bin', 'clang-format',
-        )
-        assert os.path.exists(self.binary), \
-            'Missing clang-format in {}'.format(self.binary)
+        self.diff = ''
 
     @stats.api.timed('runtime.clang-format')
     def run(self, revision):
         '''
-        Run clang-format on those modified files
+        Run ./mach clang-format on the current patch
         '''
         assert isinstance(revision, Revision)
         issues = []
-        for path in revision.files:
-
-            # Check file extension is supported
-            _, ext = os.path.splitext(path)
-            if ext not in settings.cpp_extensions:
-                logger.info('Skip clang-format for non C/C++ file', path=path)
-                continue
 
-            # Build issues for modified file
-            issues += self.run_clang_format(path, revision)
+        # Commit the current revision for `./mach clang-format` to reformat its changes
+        client = hglib.open(settings.repo_dir)
```

Comment:
You should do that in the revisions `apply` method, replacing the `nocommit` option